util/types/datum_test.go

```diff
@@ -131,6 +133,15 @@ func testDatumToInt64(c *C, val interface{}, expect int64) {
 	c.Assert(b, Equals, expect)
 }
 
+func testDatumToInt64WithTargetType(c *C, val interface{}, tp byte, expectErr Checker, expectVal int64) {
```

Comment:
you can add this test it in util/types/convert_test.go, 
thus `ToInt64WithTp `, `testDatumToInt64WithTargetType` and `testDatumToInt64WithTargetType` can be removed.